File: No_2110/Number of Smooth Descent Periods of a Stock.cpp
class Solution {
public:
    long long getDescentPeriods(vector<int>& prices) {
        vector<long long> dp(prices.size());
        dp[0] = 1;
        for(int i=1; i<prices.size(); i++) {
            if(prices[i]+1 == prices[i-1]) {
                dp[i] = dp[i-1] + 1;
            } else {
                dp[i] = 1;
            }
        }

        return accumulate(dp.begin(), dp.end(), 0LL);       
    }
};

File: No_2110/Number of Smooth Descent Periods of a Stock.py
class Solution:
    def getDescentPeriods(self, prices: List[int]) -> int:
        ans = 1
        left = 0
        for right in range(1, len(prices)):
            if prices[right] != prices[right-1] - 1:
                left = right
            
            ans += right - left + 1

        return ans